### PR TITLE
*: Fix issue #1071

### PR DESCRIPTION
**Pull request by zimulala**
fix issue #1071

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -208,6 +209,33 @@ func (e *CheckTableExec) Close() error {
...
+func (e *TableDualExec) Fields() []*ast.ResultField {
+	return e.fields
+}
+
+// Next implements Execution Next interface.
+func (e *TableDualExec) Next() (*Row, error) {
+	if e.executed {
+		return nil, nil
+	}
+
```
Comment (coocood) [inline executor/executor.go]: remove empty line

File: executor/executor.go [hunk 2]
```diff
@@ -208,6 +209,33 @@ func (e *CheckTableExec) Close() error {
...
+	e.executed = true
+
```
Comment (coocood) [inline executor/executor.go]: same above

File: optimizer/plan/planbuilder.go [hunk 3]
```diff
@@ -324,6 +327,14 @@ func (b *planBuilder) buildAllAccessMethodsPlan(path *joinPath) []Plan {
 	return candidates
 }
 
+func (b *planBuilder) buildTableDual(sel *ast.SelectStmt) Plan {
+	dual := &TableDual{FilterConditions: splitWhere(sel.Where)}
+	ret := ast.ResultField{}
+	dual.SetFields([]*ast.ResultField{&ret})
+
```
Comment (coocood) [inline optimizer/plan/planbuilder.go]: remove empty line

**Comment by coocood** [comment]
LGTM

File: executor/executor.go [hunk 4]
```diff
@@ -208,6 +209,31 @@ func (e *CheckTableExec) Close() error {
...
+	fields   []*ast.ResultField
+	executed bool
+}
+
+// Fields implements Executor Fields interface.
+func (e *TableDualExec) Fields() []*ast.ResultField {
+	return e.fields
+}
+
+// Next implements Execution Next interface.
```
Comment (shenli) [inline executor/executor.go]: Execution -> Executor

File: optimizer/plan/plans.go [hunk 5]
```diff
@@ -26,6 +26,21 @@ type TableRange struct {
...
+	basePlan
+
+	HasAgg bool
+	// FilterConditions can be used to filter result.
+	FilterConditions []ast.ExprNode
+}
+
+// Accept implements Plan Accept interface.
+func (p *TableDual) Accept(v Visitor) (Plan, bool) {
+	np, _ := v.Enter(p)
```
Comment (shenli) [inline optimizer/plan/plans.go]: Apply visitor on FilterConditions?

**Comment by shenli** [comment]
LGTM

**Pull request merged**